File: sol/solution/0301-0400/0303/Solution.ts
class NumArray {
    private prefixSum: number[];
    constructor(nums: number[]) {
        this.prefixSum = [0];
        for (let i = 0; i < nums.length; i++) {
            this.prefixSum[i + 1] = this.prefixSum[i] + nums[i];
        }
    }
    sumRange(left: number, right: number): number {
        return this.prefixSum[right + 1] - this.prefixSum[left];
    }
}
/**
 * Your NumArray object will be instantiated and called as such:
 * var obj = new NumArray(nums)
 * var param_1 = obj.sumRange(left,right)
 */

File: sol/solution/0301-0400/0303/Solution.cpp
class NumArray {
private:
   vector<int> prefixsum ;
public:
    NumArray(vector<int>& nums) {
        int sum = 0 ;
        prefixsum.resize(nums.size()) ;
        for ( int i= 0 ; i<nums.size() ;i++ ){
          sum += nums[i] ;
          prefixsum[i] = sum ;
        }
    }
    int sumRange(int left, int right) {
        if ( left == 0) return prefixsum[right] ;
        return prefixsum[right]-prefixsum[left-1] ;
    }
};
/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray* obj = new NumArray(nums);
 * int param_1 = obj->sumRange(left,right);
 */